homeassistant/components/pvoutput/config_flow.py

```diff
@@ -90,3 +91,49 @@ async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
                 CONF_API_KEY: config[CONF_API_KEY],
             }
         )
+
+    async def async_step_reauth(self, data: dict[str, Any]) -> FlowResult:
+        """Handle initiation of re-authentication with PVOutput."""
+        self.reauth_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle re-authentication with PVOutput."""
+        errors = {}
+
+        if user_input is not None and self.reauth_entry:
+            try:
+                await validate_input(
+                    self.hass,
+                    api_key=user_input[CONF_API_KEY],
+                    system_id=self.reauth_entry.data[CONF_SYSTEM_ID],
+                )
+            except PVOutputAuthenticationError:
+                errors["base"] = "invalid_auth"
+            except PVOutputError:
+                errors["base"] = "cannot_connect"
+            else:
+                self.hass.config_entries.async_update_entry(
+                    self.reauth_entry,
+                    data={
+                        **self.reauth_entry.data,
+                        CONF_API_KEY: user_input[CONF_API_KEY],
+                    },
+                )
+                self.hass.async_create_task(
+                    self.hass.config_entries.async_reload(self.reauth_entry.entry_id)
+                )
+                return self.async_abort(reason="reauth_successful")
```

Comment:
We're missing an abort string for reauth_successful.

Reply:
Oh lord, I didn't include the translations 🤦 

Well spotted, thanks! 👍